File: ruby/spec/year2016/sum_square_difference_spec.rb
require_relative "../../src/year2016/sum_square_difference"

RSpec.describe "sum square difference" do

  it "finds sum square difference of 0 numbers" do
    expect(SumSquareDifference.new.findDifference(0).to eq(0))
  end

  it "finds sum square difference of 2 numbers" do
    expect(SumSquareDifference.new.findDifference(2).to eq(4))
  end

  it "finds sum square difference of 10 numbers" do
    expect(SumSquareDifference.new.findDifference(10).to eq(2640))
  end

end

File: javascript/src/test/year2016/sumSquareDifference.spec.js
'use strict';

require('../../main/year2016/sumSquareDifference.js');
var expect = require('chai').expect;

describe("sum square difference", function () {

    it("finds sum square difference of 0 numbers", function () {
        expect(sumSquareDifference(0)).to.eql(0);
    });

    it("finds sum square difference of 2 numbers", function () {
        expect(sumSquareDifference(2)).to.eql(4);
    });

    it("finds sum square difference of 10 numbers", function () {
        expect(sumSquareDifference(10)).to.eql(2640);
    });

});
